src/Server/MySQLHandler.cpp

```diff
@@ -472,14 +472,10 @@ void MySQLHandler::comQuery(ReadBuffer & payload, bool binary_protocol)
         auto query_context = session->makeQueryContext();
         query_context->setCurrentQueryId(fmt::format("mysql:{}:{}", connection_id, toString(UUIDHelpers::generateV4())));
 
-        /// --- Workaround for Bug 56173. Can be removed when the analyzer is on by default.
```

Comment:
Should we remove it, if still for many people analyzer it not turned on by default?

Reply:
There are unfortunately more complications (--> https://github.com/ClickHouse/support-escalation/issues/2848#issuecomment-2358075429). These need to be clarified before this PR is continued.

But your point is a good one ... the removal could/should be made dependent on analyzer = 1 vs. =0.